### ISSUE TITLE
Is there a way to remove a run using the client library?

### ISSUE THREAD

**Issue by wjaskowski** [neptune-ai/neptune-client#610]
Is there a way to remove/delete/trash a run using the client library? Alternatively, completely wipe out (remove all the data) an existing run?

**Comment by Herudaio** [comment]
Hi @wjaskowski !

Currently there is no option to programmatically delete or trash runs. We do plan to add project level functionalities to the neptune-client, but it's not scoped out and I cannot promise a specific timeline.

To wipe out run completely you can:
• Use `run.get_structure()` to get the run structure that you can programmatically navigate through, then on all non-sys fields invoke `run.pop(path)` or `del run[path]`
• Or soon (this is already in backlog for development) you would be able to invoke `del`/`pop` on namespace level deleting all deeper level fields.

**Comment by robmoore** [comment]
@Herudaio Just curious if the second option you mentioned using `del` / `pop` is available now?

**Comment by Herudaio** [comment]
@robmoore yes it is :) You can invoke `del` on namespaces and it will remove all fields and their data in that namespace or deeper within.

**Comment by wjaskowski** [comment]
But still no way to trash a run programmatically, right? I would not close the issue then.

**Comment by Herudaio** [comment]
@wjaskowski there is [an open feature request about deleting the run programatically](https://github.com/neptune-ai/neptune-client/issues/727). As this question mentioned either of two features and one is already added I took the liberty of closing the issue :)
